src/drivers/sqs/Queue.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace yii\queue\sqs;
+
+use yii\base\NotSupportedException;
+use yii\queue\cli\Queue as CliQueue;
+use yii\queue\cli\Signal;
+use \Aws\Sqs\SqsClient;
+use Aws\Credentials\CredentialProvider;
+
+/**
+ * SQS Queue
+ *
+ * @author Manoj Malviya <manojm@girnarsoft.com>
+ */
+class Queue extends CliQueue
+{
+    /**
+     * The SQS url.
+     * @var string
+     */
+    public $url;
+
+    public $key = '';
+    public $secret = '';
+    public $region = '';
+    public $version = 'latest';
+
+    /**
+     * @var string command class name
+     */
+    public $commandClass = Command::class;
+
+    /**
+     * @inheritdoc
+     */
+     public function init()
+     {
+         parent::init();
+     }
+
+    /**
+     * Runs all jobs from queue.
+     */
+    public function run()
+    {
+        while (!Signal::isExit() && ($payload = $this->getPayload())) {
+            list($ttr, $message) = explode(';', $payload['Body'], 2);
+
+            $this->reserve($payload, $ttr); //reserve it so it is not visible to another worker till ttr
+
+            if($this->handleMessage(null, $message, $ttr, 1))
+            {
+                //if handled then remove from queue
+                $this->release($payload);
+            }
+        }
+    }
+
+    /**
+     * Listens to get new jobs.
+     */
+    public function listen()
+    {
+        $this->run();        
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function pushMessage($message, $ttr, $delay, $priority)
```

Comment:
Is it method is used anywhere ?

Reply:
As I know it's used here 
https://github.com/yiisoft/yii2-queue/blob/967b0e149ba40a7af7ed55080913a5c264164661/src/Queue.php#L146

Reply:
sorry, i didn't notice.